Source/WebKit/UIProcess/API/wpe/WPEWebViewPlatform.cpp

```diff
@@ -134,10 +135,16 @@ ViewPlatform::ViewPlatform(WPEDisplay* display, const API::PageConfiguration& co
 
     auto& pageConfiguration = m_pageProxy->configuration();
     m_pageProxy->initializeWebPage(pageConfiguration.openedSite(), pageConfiguration.initialSandboxFlags());
+
+    WebCore::SystemSettings::singleton().addObserver([this](const auto& state) {
+        if (state.darkMode)
```

Comment:
I understand the logic in `effectiveAppearanceIsDark()`, returning `value_of(false)`. 

But here, what happens if we were in dark mode and we turn to light mode? this check would fail so we wouldn't call `didChange`. Is the transition from dark to light handled elsewhere?

Reply:
The `std::optional` means that no value has been explicitly set in the settings, so the `.value_or(false)` is to disable by default dark mode if a value has not been set.

Reply:
Ah! This `darkMode` is an optional! Then yes, the test is true when a value (true or false) is set, I see :-)